traveler-ios-ui/TravelerKitUI/ViewControllers/CancellationViewController.swift

```diff
@@ -15,59 +15,134 @@ protocol CancellationViewControllerDelegate: class {
 }
 
 class CancellationViewController: UITableViewController {
-    @IBOutlet weak var totalRefundLabel: UILabel!
+    @IBOutlet weak var explanationInputTextField: UITextField!
 
     var quote: CancellationQuote?
     weak var delegate: CancellationViewControllerDelegate?
 
+    private var selectedReason: CancellationReason?
+    private var explanation: String?
+    private var selectedReasonIndexPath: IndexPath?
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        totalRefundLabel.text = quote?.totalRefund.localizedDescriptionInBaseCurrency
+        explanationInputTextField.delegate = self
```

Comment:
You should use events instead of delegates here. The UITextFieldDelegate is used for something else